github-action publishing uploads master package as well (i.e. python/ffmpeg)

### Is this a new Bug?

- [X] I checked that the bug hasn't been reported before

### Package Name

deluge

### Package Version

any

### Device Model

any

### Device Architecture

x86_64

### Firmware Version

unrelated

### What happened?

Invoking builds + upload through github-action for shared packages (i.e. using python311 of ffmpeg) ends-up uploading both artifacts instead of the resulting package being built.  This ends-up generating errors:
```
SUCCESS:
2023.09.07 01:35:03 - sabnzbd: (evansport-7.1) DONE

ARTIFACTS:
-rw-r--r-- 1 root  27M 2023.09.07 01:31:30 packages/python311_evansport-7.1_3.11.5-8.spk
-rw-r--r-- 1 root 6.3M 2023.09.07 01:35:01 packages/sabnzbd_evansport-7.1_4.0.3-65.spk


UNSUPPORTED (skipped):
none.

Error: ERRORS:
2023.09.07 01:31:33 - python311: (evansport-7.1) FAILED

See log file of the build job to analyze the error(s).

Last [15](https://github.com/th0ma7/spksrc/actions/runs/6103372968/job/16563729213#step:7:16) lines of the build log:


$ make publish-arch-evansport-7.1 -C ./spk/python311
/github/workspace/mk/wizard/uninstall_uifile
(cd /github/workspace/spk/python311/work-evansport-7.1 && tar cpf /github/workspace/spk/python311/../../packages/python311_evansport-7.1_3.11.5-8.spk --group=root --owner=root package.tgz INFO scripts conf PACKAGE_ICON.PNG PACKAGE_ICON_256.PNG WIZARD_UIFILES)
ERROR: Failed to publish package - HTTP response code 409
HTTP/1.1 409 CONFLICT
Server: nginx/1.25.1
Date: Thu, 07 Sep [20](https://github.com/th0ma7/spksrc/actions/runs/6103372968/job/16563729213#step:7:21)[23](https://github.com/th0ma7/spksrc/actions/runs/6103372968/job/16563729213#step:7:24) 01:[31](https://github.com/th0ma7/spksrc/actions/runs/6103372968/job/16563729213#step:7:32):[33](https://github.com/th0ma7/spksrc/actions/runs/6103372968/job/16563729213#step:7:34) GMT
Content-Type: application/json
Content-Length: 52
Connection: keep-alive

{"message": "Conflicting architectures: evansport"}
make[1]: *** [../../mk/spksrc.spk.mk:481: publish] Error 1
make[1]: Leaving directory '/github/workspace/spk/python311'
make: *** [../../mk/spksrc.spk.mk:709: publish-arch-evansport-7.1] Error 1
make: Leaving directory '/github/workspace/spk/python311'
```

### Reproduction steps

n/a

### Install Log

```log
n/a
```


### Service Log

_No response_

### Other Logs

_No response_

--- a/.github/actions/build.sh
+++ b/.github/actions/build.sh
@@ -66,8 +66,14 @@ do
     echo >build.log
 
     if [ "${GH_ARCH%%-*}" != "noarch" ]; then
-        echo "$ make ${MAKE_ARGS}arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package}" >>build.log
-        make ${MAKE_ARGS}arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package} |& tee >(tail -15 >>build.log)
+        # Only publish package requested and not its parent (i.e. python*, ffmpeg*)
+        if [ "${package}" == "${SPK_TO_BUILD}" ]; then
+           echo "$ make ${MAKE_ARGS}arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package}" >>build.log
+           make ${MAKE_ARGS}arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package} |& tee >(tail -15 >>build.log)
+        else
+           echo "$ make arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package}" >>build.log
+           make arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package} |& tee >(tail -15 >>build.log)
+        fi
     else
         if [ "${GH_ARCH}" = "noarch" ]; then
             TCVERSION=

--- a/.github/actions/prepare.sh
+++ b/.github/actions/prepare.sh
@@ -18,8 +18,8 @@ make setup-synocommunity
 DEFAULT_TC=$(grep DEFAULT_TC local.mk | cut -f2 -d= | xargs)
 
 # filter for changes made in the spk directories and take unique package name (without spk folder)
-SPK_TO_BUILD+=" "
-SPK_TO_BUILD+=$(echo "${GH_FILES}" | tr ' ' '\n' | grep -oP "(spk)/\K[^\/]*" | sort -u | tr '\n' ' ')
+SPKLIST_TO_BUILD="${SPK_TO_BUILD} "
+SPKLIST_TO_BUILD+=$(echo "${GH_FILES}" | tr ' ' '\n' | grep -oP "(spk)/\K[^\/]*" | sort -u | tr '\n' ' ')
 
 # filter for changes made in the cross and native directories and take unique package name (including cross or native folder)
 DEPENDENT_PACKAGES=$(echo "${GH_FILES}" | tr ' ' '\n' | grep -oP "(cross|native)/[^\/]*" | sort -u | tr '\n' ' ')
@@ -39,22 +39,22 @@ do
     echo "===> Searching for dependent package: ${package}"
     packages=$(echo "${DEPENDENCY_LIST}" | grep " ${package} " | grep -o ".*:" | tr ':' ' ' | sort -u | tr '\n' ' ')
     echo "===> Found: ${packages}"
-    SPK_TO_BUILD+=${packages}
+    SPKLIST_TO_BUILD+=${packages}
 done
 
 # fix for packages with different names
-if [ "$(echo ${SPK_TO_BUILD} | grep -ow nzbdrone)" != "" ]; then
-    SPK_TO_BUILD=$(echo "${SPK_TO_BUILD}" | tr ' ' '\n' | grep -v "nzbdrone" | tr '\n' ' ')" sonarr3"
+if [ "$(echo ${SPKLIST_TO_BUILD} | grep -ow nzbdrone)" != "" ]; then
+    SPKLIST_TO_BUILD=$(echo "${SPKLIST_TO_BUILD}" | tr ' ' '\n' | grep -v "nzbdrone" | tr '\n' ' ')" sonarr3"
 fi
-if [ "$(echo ${SPK_TO_BUILD} | grep -ow python)" != "" ]; then
-    SPK_TO_BUILD=$(echo "${SPK_TO_BUILD}" | tr ' ' '\n' | grep -v "python" | tr '\n' ' ')" python2"
+if [ "$(echo ${SPKLIST_TO_BUILD} | grep -ow python)" != "" ]; then
+    SPKLIST_TO_BUILD=$(echo "${SPKLIST_TO_BUILD}" | tr ' ' '\n' | grep -v "python" | tr '\n' ' ')" python2"
 fi
-if [ "$(echo ${SPK_TO_BUILD} | grep -ow ffmpeg)" != "" ]; then
-    SPK_TO_BUILD=$(echo "${SPK_TO_BUILD}" | tr ' ' '\n' | grep -v "ffmpeg" | tr '\n' ' ')" ffmpeg4"
+if [ "$(echo ${SPKLIST_TO_BUILD} | grep -ow ffmpeg)" != "" ]; then
+    SPKLIST_TO_BUILD=$(echo "${SPKLIST_TO_BUILD}" | tr ' ' '\n' | grep -v "ffmpeg" | tr '\n' ' ')" ffmpeg4"
 fi
 
 # remove duplicate packages
-packages=$(printf %s "${SPK_TO_BUILD}" | tr ' ' '\n' | sort -u | tr '\n' ' ')
+packages=$(printf %s "${SPKLIST_TO_BUILD}" | tr ' ' '\n' | sort -u | tr '\n' ' ')
 
 # for ffmpeg v4-6 find all packages that depend on them
 for i in {4..6}; do

